### ISSUE TITLE
Identify already-cleaned, must-have SOLR fields

### ISSUE THREAD

**Issue by davidskalinder** [davidskalinder/mpeds-coder#75]
#59 has a lot on this already, and some more in #70.  Opening a separate issue here mostly to keep better track of it.

EDIT: I'm also going to paste here from #59 the list of goals for these fields since I keep linking to them in this issue:

> 1. Ensure that MAI and subsequent passes have the data necessary to code and analyze
> 2. Be able to find any article in SOLR
> 3. Be able to find any article within the files that were loaded into SOLR
> 4. Be able to request any article from an external database provider (ProQuest/LexisNexis/etc.)
> 5. Be able to request any article from the original publication
> 6. Be able to cite any article within an academic bibliography

**Comment by davidskalinder** [comment]
(Moved this info from #70:)

These fields almost certainly include `DATE`, `DOCSOURCE`, `id`, `PUBLICATION`, `TEXT`, and `TITLE` (some of which are already in MAI).  `_version_` also seems pretty consistent within SOLR, but I'm not sure where it comes from...  And there might be other important ones too (`Database`?).  For more details, see #59 and the spreadsheet discussed there.

The imported fields should provide enough information to:

1. Replicate existing MAI functionality
2. Make MAI aware of publication (in order to handle coder assignments in #58)

And a little more aspirationally:

3. Identify any article within the files that were loaded into SOLR

**Comment by davidskalinder** [comment]
`_version_` seems to be a deeply internal SOLR tool for handling document concurrency and updates: 

https://stackoverflow.com/questions/15527380/solr-4-2-what-is-version-field/15581292#15581292

http://yonik.com/solr/optimistic-concurrency/

I guess the idea is SOLR keeps track of what version of the document it has in case it gets updated.  Anyway, I think we can safely omit this from any non-SOLR systems.

**Comment by davidskalinder** [comment]
Next to track down: `Database`

**Comment by davidskalinder** [comment]
Okay, so apparently `Database` is a field that we get from some data providers (from Proquest for sure, and probably LexisNexis, but apparently not from, e.g. Gigaword).  Presumably these providers use it to keep track of which of their products the article comes from.  It looks like @alexhanna might have also used it to label some of the web scraping batches that the campus protest project did.

Here's an example from a Proquest export:

```
Copyright: Copyright Caribbean Today Apr 30, 1991

Last updated: 2012-02-08

Database: Ethnic NewsWatch
```

So in terms of the list of goals at https://github.com/davidskalinder/mpeds-coder/issues/59#issuecomment-611150228, the `Database` field only helps with the 4th (being able to request an article from the database provider).  Even for that task, my guess is that some other field (like a provider-specific ID) will accomplish it more efficiently.

So tl;dr: I'm pretty confident that the `Database` field is not essential for this issue (and maybe not even for #74).

**Comment by davidskalinder** [comment]
Next up: `Internal ID`

**Comment by davidskalinder** [comment]
Well, based on what @alexhanna told me last week, and having done some further digging, it looks like `INTERNAL_ID` is sort of a best-guess locator that _might_ help to find an article within the file that was received from the data provider.  Usually the only information the field contains that isn't found anywhere else is the first integer of a source file's line like `Document 311 of 1000`; but for some data sources there isn't an `INTERNAL_ID` at all.

So: it looks like the only item of the list above that this field could help would be goal 3 (finding the article within the pre-SOLR download file); and I'm not convinced that it even adds much value beyond, say, the article title.  So my inclination is to not worry about this field for this issue, and I'm dubious whether it will be worth including in #74 either.

**Comment by davidskalinder** [comment]
Aside from the ones above, it looks like most of the fields that @olderwoman has marked as being necessary are stored too inconsistently in SOLR to import safely to MAI without some further cleaning (I marked these in the spreadsheet on the Google Drive, which I've now moved to the folder `Technical documentation > SOLR_documentation`); so they will need to be handled by #72 and #74.

The list of remaining fields is:

```
DATE
DOCSOURCE
id
PUBLICATION
TEXT
TITLE
```

The MAI table already has `DATE`, `id`, and `TITLE` (as `article_metadata`'s columns `pub_date`, `db_id`, and `title` respectively).

So, after all that, it looks like the fields that need to be ported over are `PUBLICATION`, `DOCSOURCE`, and `TEXT`.  The first two should be pretty easy, though of course the full text needs more caution.

**Comment by davidskalinder** [comment]
Tl;dr version for @alexhanna and/or @olderwoman: do we remember whether SOLR's `AP`, `UPI`, `LENGTH`, and `SEARCH_ID` fields ever provide anything useful enough that we'd want to display them to coders atop the article text in the future?

Long version: 

Looking a little more carefully at the MAI page that displays the article though (prepared by the controller functions `prepText()` and `loadSolr()`), I realize that there is one other consideration for replicating the current functionality without SOLR.  The following fields seem to be opportunistically loaded into the metadata line at the top of the article display whenever they are available:


displayed_field | in_any_live_dbs | bpp_usage | DS_notes
-- | -- | -- | --
AP | FALSE |   |  
BYLINE | TRUE |   | Inconsistent, will need cleaning
By | FALSE |   |  
DATE | TRUE | 6453 | Include
DATELINE | TRUE |   | Inconsistent, will need cleaning
EDITION | TRUE |   | Inconsistent, will need cleaning
INTERNAL_ID | TRUE | 6453 |  
LANGUAGE | TRUE |   |  
LENGTH | TRUE |   |  
PUBLICATION | TRUE | 6453 | Include
Published | FALSE |   |  
SEARCH_ID | FALSE |   |  
SECTION | TRUE | 1373 | Inconsistent, will need cleaning
UPI | FALSE |   |  

So the only one of these that we're not planning to port that might ever appear in the current BPP production data is `SECTION` (and indeed I'm not even sure about that one, since it's `Section` in the BPP data and I don't know whether that would match).

Of the others, we're already going to consider cleaning the most obviously useful ones; `INTERNAL_ID` and `Published` are likely covered by `db_id` and `pub_date`; `By` will be rolled into `BYLINE`; and it's hard to imagine why we'd need to see `LANGUAGE` displayed with the article text.  `LENGTH` (a word count that sometimes appears in the campus protest data) seems similarly likely to be useless?

That leaves `AP`, `UPI`, and `SEARCH_ID`, which I guess I'd like to know more about before deciding to either relegate their handling to #74 or to plan to abandon them altogether.  Maybe @olderwoman or @alexhanna could advise on this?

Otherwise, I think the triage plan in the comments above is still the best way to proceed?...

**Comment by davidskalinder** [comment]
I've now ported `PUBLICATION` and `DOCSOURCE` to all our MAI deployments, and `TEXT` will be handled in separate issues.  The other suggestions in the comment above don't seem to have generated any fury, so I'm going to close this issue for now (though of course we can reopen later if necessary).
